index.bs

```diff
@@ -14602,11 +14609,14 @@ Each {{DOMException}} object has an associated <dfn for="DOMException">name</dfn
 <dfn for="DOMException">message</dfn>, both [=strings=].
 
 The
-<dfn constructor for="DOMException" lt="DOMException(message, name)"><code>new DOMException(|message|, |name|)</code></dfn>
+<dfn constructor for="DOMException" lt="DOMException(message, options)"><code>new DOMException(|message|, |options|)</code></dfn>
 constructor steps are:
 
-1. Set [=this=]'s [=DOMException/name=] to |name|.
-1. Set [=this=]'s [=DOMException/message=] to |message|.
+1.  Set [=this=]'s [=DOMException/message=] to |message|.
+1.  If |options| is a string, then set [=this=]'s [=DOMException/name=] to |options|.
+1.  Otherwise,
+    1.  Set [=this=]'s [=DOMException/name=] to |options|["{{DOMExceptionOptions/name}}"].
+    1.  Perform [=?=] <a abstract-op>InstallErrorCause</a>([=this=], |options|).
```

Comment:
It's not good that InstallErrorCause() performs HasProperty() and Get(), which the Web IDL bindings layer has already done.

It's also not really valid spec text to pass |options|, which is a Web IDL dictionary at this point, to InstallErrorCause(), which expects a JavaScript object. (Although, we do often fudge these boundaries; I've certainly done it myself.)

I think we should instead duplicate the InstallErrorCause logic here:

2. If |options|["cause"] [=map/exists=], then perform [CreateNonEnumerableDataPropertyOrThrow](https://tc39.es/ecma262/#sec-createnonenumerabledatapropertyorthrow)(O, "cause", |options|["cause"]).

Reply:
Thank you for the suggestion!

AFAICT, `O` is not defined in this context. Should we perform abstract-op CreateNonEnumerableDataPropertyOrThrow on `[=this=]` instead? I'm not sure if `[=this=]` here refers to the WebIDL type or the JavaScript `this` 😅

Reply:
You can use `[=this=]` and assume it refers to an ES object. Technically it is defined to refer to an ES object, but in general we have let it be used both ways in some cases like this. (My personal rule is: if the IDL value type is `any` or `object`, then you can assume implicit conversions, since the conversions are trivial. That is why I am less comfortable with `|options|`, since it is a dictionary type and dictionary conversions are not trivial.)